src/grid/grid_task_list.c

```diff
@@ -121,46 +201,47 @@ void grid_free_task_list(grid_task_list *task_list) {
  * \author Ole Schuett
  ******************************************************************************/
 void grid_collocate_task_list(
-    const grid_task_list *task_list, const bool orthorhombic, const int func,
-    const int nlevels, const int npts_global[nlevels][3],
-    const int npts_local[nlevels][3], const int shift_local[nlevels][3],
-    const int border_width[nlevels][3], const double dh[nlevels][3][3],
-    const double dh_inv[nlevels][3][3], double *grid[nlevels]) {
+    const grid_task_list *task_list, const bool orthorhombic,
+    const enum grid_func func, const int nlevels,
+    const int npts_global[nlevels][3], const int npts_local[nlevels][3],
+    const int shift_local[nlevels][3], const int border_width[nlevels][3],
+    const double dh[nlevels][3][3], const double dh_inv[nlevels][3][3],
+    double *grid[nlevels]) {
 
   // If validation is enabled, make a backup copy of the original grid.
-  double *grid_before[nlevels]; // Only used for validation.
-  if (task_list->validate) {
-    for (int level = 0; level < nlevels; level++) {
-      const size_t sizeof_grid = sizeof(double) * npts_local[level][0] *
-                                 npts_local[level][1] * npts_local[level][2];
-      grid_before[level] = malloc(sizeof_grid);
-      memcpy(grid_before[level], grid[level], sizeof_grid);
-      memset(grid[level], 0, sizeof_grid);
-    }
-  }
```

Comment:
Why did you remove this part?

Reply:
because anytime the function containing this code is called the grid is always set to zero before (it is done on the fortran side). So the copy to temporary array, then the second zeroing of the grid is not needed. You can just apply the collocate function to the grid and then call the reference backend to the second temporary space you allocate after and do the comparison.